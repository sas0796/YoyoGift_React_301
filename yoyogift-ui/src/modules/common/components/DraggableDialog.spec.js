import React from "react";
import { mount } from "enzyme";

import DraggableDialog from "./DraggableDialog";

describe("Draggable Dialog Test Suite", () => {
  it("should render expected HTML", () => {
      expect(mount(<DraggableDialog />).html()).toMatchSnapshot();
  });

  it('should call handle open close function', () => {
    //Mocks
    let classes = {root: {marginTop: "1px"}};

    const wrapper = mount(<DraggableDialog />)
    const component = wrapper.instance();
    const spy =  jest.spyOn(component, 'handleOpenClose');
    
    wrapper.find("button")
    .at(0)
    .props()
    .onClick();

    component.handleOpenClose();
   
   expect(component.handleOpenClose).toHaveBeenCalled();
   expect(component.handleOpenClose).toHaveBeenCalledTimes(1)
  });

  it('should call handle open close function', () => {
    //Mocks
    let classes = {root: {marginTop: "1px"}};

    const wrapper = mount(<DraggableDialog getEmail={jest.fn()}/>)
    const component = wrapper.instance();
    const spy =  jest.spyOn(component, 'handleSendAndClose');
    
    wrapper.find("button")
    .at(0)
    .props()
    .onClick();

    component.handleSendAndClose("abc");
   
   expect(component.handleSendAndClose).toHaveBeenCalled();
   expect(component.handleSendAndClose).toHaveBeenCalledTimes(1)
  });

});
  
