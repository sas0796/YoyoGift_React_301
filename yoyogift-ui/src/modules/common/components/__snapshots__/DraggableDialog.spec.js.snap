// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Draggable Dialog Test Suite should render expected HTML 1`] = `"<div><button class=\\"MuiButtonBase-root-27 MuiButton-root-1 MuiButton-contained-12 MuiButton-containedPrimary-13 MuiButton-raised-15 MuiButton-raisedPrimary-16\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label-2\\">send this gift card</span><span class=\\"MuiTouchRipple-root-46\\"></span></button></div>"`;
